docs/why.qmd

```diff
@@ -2,24 +2,224 @@
 title: "Why Ibis?"
 ---
 
+Ibis defines a Python dataframe API that supports any query engine -- the
+frontend for any backend data platform. This includes 20+ backends today, most
```

Comment:
There are two main values that can be attractive to different audiences (1. people who care about Python df API and 2. people who care about connecting to various backends). If we articulate the two values independently that may help people to easily understand. 
(also wondering if there is a way we can capture reader statistics to understand who our audience is and what specific value they care more)?

Reply:
follow up for this: https://github.com/ibis-project/ibis/issues/8514